xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.h

```diff
@@ -146,7 +148,7 @@ class CDVDDemuxFFmpeg : public CDVDDemux
 
   CCriticalSection m_critSection;
   std::map<int, CDemuxStream*> m_streams;
-  std::vector<std::map<int, CDemuxStream*>::iterator> m_stream_index;
+  std::map<int, std::map<int, CDemuxStream*>::iterator> m_stream_index;
```

Comment:
a map is complex and a lookup is slow. for a small number of items a vector is much faster when being iterated and compared. I think we should avoid this double map and get rid of the other map too.